src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/TargetMigrationOrchestrator.cs

```diff
@@ -588,19 +599,43 @@ private async Task<IMigrationPhaseWorkload> NextWorkloadRunnerAsync(PhaseResult
         {
             var endSN = await this.GetEndSequenceNumberAsync(cancellationToken);
             var delta = endSN - currentResult.EndSeqNum;
+
             if (currentResult.Phase == MigrationPhase.Catchup)
             {
-                if (delta < this.MigrationSettings.DowntimeThreshold
-                    && this.downtimeAllowed == true)
+                if (!this.IsAutoStartMigration())
```

Comment:
Simplify the logic
1) set downtimeallowed(local to method) = true when user triggered downtime api for manual mode or delta met threshold for auto mode
2) Common logic to invoke downtime phase if downtime is set, catchup otherwise

- Also remove downtimeAllowed class variable

Reply:
Cleaned up the logic as suggested